addon/components/sl-split-grid.js

```diff
@@ -0,0 +1,564 @@
+    // -------------------------------------------------------------------------
+    // Attributes
+
+    /**
+     * HTML tag name of the root element
+     *
+     * @property {string} tagName
+     * @default  "div"
+     */
+    tagName: 'div',
+
+    /**
+     * Class names for the root element
+     *
+     * @property {array} classNames
+     */
+    classNames: [ 'sl-split-grid' ],
+
+    /**
+     * Class name bindings for the root element
+     *
+     * @property {array} classNameBindings
+     */
+    classNameBindings: [
+        'detailsOpen:details-open',
+        'isLoading:sl-loading',
+        'pendingData:pending-data'
+    ],
+
+    // -------------------------------------------------------------------------
+    // Actions
+
+    /**
+     * Component actions hash
+     *
+     * @property {object} actions
+     */
+    actions: {
+
+        /**
+         * Close the details pane
+         *
+         * @function actions.closeDetailsPane
+         * @returns  {void}
+         */
+        closeDetailsPane: function() {
+            var activeRecord = this.get( 'activeRecord' );
+
+            if ( activeRecord ) {
+                Ember.set( activeRecord, 'active', false );
+                this.set( 'activeRecord', null );
+            }
+
+            this.set( 'detailsOpen', false );
+        },
+
+        /**
+         * Open the details pane with a specific row object
+         *
+         * @function actions.openDetailsPane
+         * @param    {object} row - The object that the clicked row represents
+         * @returns  {void}
+         */
+        openDetailsPane: function( row ) {
+            var activeRecord = this.get( 'activeRecord' );
+
+            if ( activeRecord ) {
+                Ember.set( activeRecord, 'active', false );
+            }
+
+            Ember.set( row, 'active', true );
+            this.set( 'activeRecord', row );
+
+            this.set( 'detailsOpen', true );
+
+            if ( this.get( 'autoHeight' ) ) {
+                Ember.run.next( this.updateDetailContentHeight.bind( this ) );
+            }
+        },
+
+        /**
+         * Toggle sorting of the selected column, and send the "sortAction"
+         * bound action the column and direction to sort
+         *
+         * @function actions.sortColumn
+         * @param    {object} column - The column definition for the triggered header's column
+         * @returns  {void}
+         */
+        sortColumn: function( column ) {
+            var columnTitle       = Ember.get( column, 'title' ),
+                sortedColumn      = this.getSortedColumn(),
+                sortedColumnTitle = this.get( 'sortedColumnTitle' ),
+                sortedDirection   = this.get( 'sortedDirection' ),
+                direction;
+
+            if ( sortedColumnTitle === columnTitle ) {
+                direction = sortedDirection === 'ascending' ? 'descending' : 'ascending';
+            } else {
+                if ( sortedColumn ) {
```

Comment:
Guess I'm looking at it from the perspective that the if statement is never needed past the initial null state you are referring to.  So it will be a check ran more often than is needed. As an aside, when I first made this comment I did not realize the sortedColumn value could be null - thought the column being sorted had already been designated as the one being interacted with.